From 741053f86eb21e6a00cca89d4b4c76720949a33f Mon Sep 17 00:00:00 2001
From: Moss Heim <moss.heim@qt.io>
Date: Tue, 14 Oct 2025 10:36:18 +0200
Subject: [PATCH 393/553] Doc: Add more info to QThread::requestInterruption

Document its behavior on main thread & non-running threads.

Pick-to: 6.9 6.8
Change-Id: I0857217effa727971fb532adf70e892d205036dd
Reviewed-by: Ulf Hermann <ulf.hermann@qt.io>
(cherry picked from commit e5b6e89bb6bcf9f3de5a55d60912864ab9525be9)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/corelib/thread/qthread.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/corelib/thread/qthread.cpp b/src/corelib/thread/qthread.cpp
index ccd944a0824..3e08df2b5ac 100644
--- a/src/corelib/thread/qthread.cpp
+++ b/src/corelib/thread/qthread.cpp
@@ -1291,6 +1291,9 @@ bool QThread::event(QEvent *event)
     This function does not stop any event loop running on the thread and
     does not terminate it in any way.
 
+    This function has no effect on the main thread, and does nothing if the thread
+    is not currently running.
+
     \sa isInterruptionRequested()
 */
 
-- 
2.51.2

